targets/go/source/PlayFabHttp.go.ejs

```diff
@@ -0,0 +1,165 @@
+	ProductionEnvironmentURL string
+	EnableCompression        bool
+	VerticalName             string
+	TitleId                  string
+	DisableSsl               bool
+	APIPath                  string
+}
+
+// NewSettings returns a pointer to a Settings struct that carries default values
+func NewSettings() *Settings {
+	return &Settings{
+		VersionString:            "GoSdk-",
+		RequestGetParams:         map[string]string{"sdk": ""},
+		ProductionEnvironmentURL: ".playfabapi.com", // This is only for customers running a private cluster.  Generally you shouldn't touch this
+		EnableCompression:        false,
+		VerticalName:             "", // The name of a customer vertical. This is only for customers running a private cluster. Generally you shouldn't touch this
+		TitleId:                  "",
+		DisableSsl:               false,
+	}
+}
+
+// GetFullURL returns the full URL to be used for the API call
+func (p *Settings) GetFullURL(apiPath string) string {
+	var output string
+
+	if !strings.HasPrefix(p.ProductionEnvironmentURL, "http") {
+		if p.VerticalName != "" {
+			output = "https://" + p.VerticalName
+		} else {
+			output = "https://" + p.TitleId
+		}
+	}
+
+	output += p.ProductionEnvironmentURL
+	output += apiPath
+
+	firstParam := true
+	for key, val := range p.RequestGetParams {
+		if firstParam {
+			output += "?"
+			firstParam = false
+		} else {
+			output += "&"
+		}
+
+		output += fmt.Sprintf("%s=%s", key, val)
+	}
+
+	return output
+}
+
+// Request executes the actual HTTP call
+func Request(p *Settings, b []byte, apiPath string, authKey string, authValue string) (map[string]interface{}, error) {
+
+	client := &http.Client{
+		Timeout: time.Second * 30, // 30 seconds timeout should be enough
+	}
+
+	if p.DisableSsl {
+		client.Transport = &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+	}
+
+	var req *http.Request
+	var err error
+
+	// compression enabled, so gzip POST data
+	if p.EnableCompression {
+		var buf bytes.Buffer
+		zw := gzip.NewWriter(&buf)
+		if _, err := zw.Write(b); err != nil {
+			return nil, err
+		}
+		if err := zw.Flush(); err != nil {
+			return nil, err
+		}
+		if err := zw.Close(); err != nil {
+			return nil, err
+		}
+		req, err = http.NewRequest("POST", p.GetFullURL(apiPath), &buf)
+		if err != nil {
+			return nil, err
+		}
+		req.Header.Add("Content-Encoding", "GZIP")
+		req.Header.Add("Accept-Encoding", "GZIP")
+	} else {
+		req, err = http.NewRequest("POST", p.GetFullURL(apiPath), bytes.NewBuffer(b))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("X-ReportErrorAsSuccess", "true")
+	req.Header.Add("X-PlayFabSDK", p.VersionString)
+
+	if authKey != "" && authValue != "" {
+		req.Header.Add(authKey, authValue)
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
```

Comment:
This should create structured data for the error when it fails. Go looks similar to the NodeSDK.. you might want to check that out here: https://github.com/PlayFab/NodeSDK/blob/master/PlayFabSdk/Scripts/PlayFab/PlayFab.js around line 96